src/server/fedavg.py

```diff
@@ -137,6 +154,20 @@ def __init__(
         ):
             self.args = parse_config_file(self.args)
         fix_random_seed(self.args.seed)
+
+        # generate the name for output dir, e.g. "tau_0.1_mu_2.5_2023-12-04-20:30:49" for MOON.
+        begin_time = str(local_time())
+        known_args, unknown_args = get_fedavg_argparser().parse_known_args()
+        args_original_dict = vars(known_args)
+        args_all_dict = vars(self.args)
+        diff_keys = args_all_dict.keys() - args_original_dict.keys()
+        hyperparameter_str = "_".join([f"{i}_{args_all_dict[i]}" for i in diff_keys])
+        self.name_4_output_dir = (
+            "_".join([hyperparameter_str, begin_time])
+            if len(hyperparameter_str) > 0
+            else begin_time
+        )
+
```

Comment:
Got it. 
I still stand for my opinion because the new feature should be general enough and the hyperparameter prefix would cause some problems with some FL methods like FedGen, which has a bunch of hyperparameters. Moreover, even most FL methods have a few hyperparams, these names are long and have no abbreviations.

Reply:
Like I said, this feature can be accepted by only keeping date in the dir name. So it is up to you. 
At my side, some suggestions:
1. Modify this feature (maybe only keep fix number of hyperparams in the dir name);
2. Keep the hyperparameter part as your personal feature and only remain the date part;
3. Remove this feature by yourself since the change of `src/server/fedavg.py` also includes other new feature (like FedSR) and all I can do is remove the remote `src/server/fedavg.py` at your forked FL-bench repo.

Reply:
I changed the name of output file to be the begin time.